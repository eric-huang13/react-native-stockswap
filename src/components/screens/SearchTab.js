import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import CompanyBoxList from './CompanyBoxList';
import UserList from './UserList.js';
import ArticleList from './ArticleList';

export class SearchTab extends Component {
  constructor(props) {
    super(props);
    this.state = {
      companies: true,
      users: false,
      news: false,
    };
  }

  render() {

    return (
      <View>
        <ScrollView>
          <Text style={style.header}>Search</Text>
          <View style={style.tabSelectorContainer}>
            <TouchableOpacity
              onPress={() =>
                this.setState({
                  companies: true,
                  users: false,
                  news: false,
                })
              }>
              <Text>Stocks</Text>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={() =>
                this.setState({
                  companies: false,
                  news: true,
                  users: false,
                })
              }>
              <Text>News</Text>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={() =>
                this.setState({
                  companies: false,
                  news: false,
                  users: true,
                })
              }>
              <Text>People</Text>
            </TouchableOpacity>
          </View>
          {this.state.companies ? (
            <CompanyBoxList navigation={this.props.navigation} />
          ) : this.state.news ? (
            <ArticleList />
          ) : this.state.users ? (
            <UserList  navigation={this.props.navigation}/>
          ) : (
            <View>
              <Text>Search Screen</Text>
              </View>


          )}
        </ScrollView>
      </View>
    );
  }
}

export default SearchTab;

const style = StyleSheet.create({
  searchInputContainer: {
    alignItems: 'center',
    padding: 4.8,
    marginTop: 0,
  },
  searchInput: {
    borderWidth: 0.5,
    width: '90%',
    fontSize: 15,
    fontStyle: 'italic',
    textAlign: 'left',
    padding: 2.5,
    opacity: 0.8,
    marginBottom: 20,
    paddingLeft: 5,
  },
  header: {
    fontSize: 19,
    fontWeight: '700',
    marginTop: 8.5,
    textAlign: 'center',
  },
  tabSelectorContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: 15,
  },
});
