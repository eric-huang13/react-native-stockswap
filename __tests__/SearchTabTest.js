import React from 'react';
import SearchTab from '../src/components/SearchTabScreens/SearchTab';
import ShallowRenderer from 'react-test-renderer/shallow'; 


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SearchTab/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});