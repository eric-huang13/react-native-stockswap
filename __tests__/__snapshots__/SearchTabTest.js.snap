// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#1d2842",
      "#1f2a45",
      "#222d47",
      "#242f4a",
      "#27324d",
      "#293450",
      "#2c3752",
      "#2e3955",
      "#313c58",
      "#333e5c",
      "#36415f",
      "#394463",
    ]
  }
  end={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  start={
    Object {
      "x": 0.1,
      "y": 0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ForwardRef(SafeAreaView)>
    <LinearGradient
      colors={
        Array [
          "#2c3752",
          "#2e3955",
          "#313c58",
          "#333e5c",
          "#36415f",
          "#394463",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      start={
        Object {
          "x": 0.1,
          "y": 0.1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 26.835766423357665,
            "fontWeight": "700",
            "marginVertical": 11.299270072992702,
            "textAlign": "center",
          }
        }
      >
        Search
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginVertical": 21.186131386861312,
            "paddingBottom": 2.8248175182481754,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "borderBottomColor": "#855cff",
                "borderBottomWidth": 4.237226277372263,
                "paddingBottom": 2.5423357664233577,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#8257FF",
                  "fontSize": 24.010948905109487,
                  "fontWeight": "bold",
                  "textShadowColor": "#8257FF",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "textShadowRadius": 50,
                }
              }
            >
              Stocks
            </Text>
          </View>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24.010948905109487,
                  "fontWeight": "bold",
                }
              }
            >
              News
            </Text>
          </View>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24.010948905109487,
                  "fontWeight": "bold",
                }
              }
            >
              People
            </Text>
          </View>
        </ForwardRef>
      </View>
    </LinearGradient>
    <CompanyBoxList />
  </ForwardRef(SafeAreaView)>
</LinearGradient>
`;
